src/ansys/fluent/core/solver/error_message.py

```diff
@@ -10,11 +10,16 @@ def closest_allowed_names(trial_name: str, allowed_names: str) -> List[str]:
 
 
 def allowed_name_error_message(
-    context: str, trial_name: str, allowed_values: Any
+    allowed_values: Any,
+    context: Optional[str] = None,
+    trial_name: Optional[str] = None,
+    message: Optional[str] = None,
 ) -> str:
     """Provide an error message with the closest names matching the 'trial_name' from
     the 'allowed_values' list."""
-    message = f"{trial_name} is not an allowed {context} name.\n"
+    if not message:
+        message = f"{trial_name} is not an allowed {context} name"
```

Comment:
Why only generate the message conditionally? This is the whole point if this function.